### ISSUE TITLE
Compatibility with Conv networks

### ISSUE THREAD

**Issue by xiaosu-zhu** [HEmile/storchastic#101]
I'm currently working with CNNs, which allow input `x` with shape `x.shape -> [N, C, H, W]`.

However, when I set `N` as the batch dim (use `storch.denote_independent(x, 0, "data")`) and perform `RELAX("z", n_samples=1)` sampling (it adds another plate), the sampled tensor becomes `z.shape -> [Z, N, C, H, W]` and is not compatible with Conv2D any more. 

Is there any way to reshape `z` to be compatible with Conv2D?

Minimal reproduction code:

```python
class RelaxSample(nn.Module):
    def __init__(self, k: int, c: int):
        super().__init__()
        self._k = k
        self._wv = nn.Parameter(torch.empty(k, c))
        self._method = RELAX("z", n_samples=1, in_dim=[k], rebar=True)

    def forward(self, latent):
        # [n, c, h, w]
        n, c, h, w = latent.shape
        # [n, h, w, c]
        q = latent.permute(0, 2, 3, 1)
        # [n, h, w, c], [k, c] -> [n, h, w, k]
        logit = torch.einsum("nhwc,kc->nhwk", q, self._wv)
        varPosterior = Categorical(logits=logit)
        # [z, n, h, w, k]
        z = self._method(varPosterior)
        # [z, n, c, h, w]
        z = torch.einsum("znhwk,kc->znhwc", z, self._wv).permute(0, 1, 4, 2, 3)
        # ********* z is not compatible with Conv2D *********
        return z
```

**Comment by HEmile** [comment]
Hi there, thank you for your question!
Unfortunately, some PyTorch code isn't so-called 'left-broadcastable', and it seems Conv2D is one of those instances. I implemented some code previously to allow non-broadcastable code to be compatible with Storchastic, but it was hidden a little too much. In https://github.com/HEmile/storchastic/commit/4a1f2c54436a07c41746db6b605f58aae320dd77 I added a helper wrapper `make_left_broadcastable` for this purpose (you can just import it from the `storch` module.

You should now be able to use (for example) `make_left_broadcastable(Conv2d(16, 33, 3))`. However, I've not tested this, so please let me know if you keep running into issues :)

**Comment by xiaosu-zhu** [comment]
Hi @HEmile, thank you for your nice and quick reply. I've tried the new wrapper (actually, adding `flatten_plates=True` in `@deterministic` wrapper does the same thing), it works fine! Well done you and your powerful framework!

**Comment by HEmile** [comment]
Glad to hear it works, and thank you for your nice comments :) Indeed, `make_left_broadcastable` just does `deterministic(fn, flatten_plates=True)`
